ee/kernel/src/sifrpc.c

```diff
@@ -148,11 +148,11 @@ int SifBindRpc(SifRpcClientData_t *cd, int sid, int mode)
 
     if (!SifSendCmd(SIF_CMD_RPC_BIND, bind, RPC_PACKET_SIZE, NULL, NULL, 0)) {
         rpc_packet_free(bind);
+        DeleteSema(cd->hdr.sema_id);
         return -E_SIF_PKT_SEND;
     }
 
     WaitSema(cd->hdr.sema_id);
-    DeleteSema(cd->hdr.sema_id);
```

Comment:
SCE 3.0.0 keeps this instead of moving `iDeleteSema` inside intr handler.

Reply:
Isn’t the same? Actually it would be better implementation mine, as it will generate smaller binary

Reply:
Actually this looks fine. Since `iSignalSema` puts the waiting thread in the ready queue the reference to the sema isn't needed anymore. Also it appears when `iDeleteSema` is called waiting on threads are automatically put in the ready queue.

Reply:
Nice!

Reply:
As long as the threads are properly queued when you delete the sema this looks fine to me.

Reply:
The only potential problem I see is that this creates a race condition on `WaitSema`. For debug EE kernel when debugging functions are not disabled, if the semaphore becomes invalid before `WaitSema` is called, it will trap.

Reply:
How can be this possible? I can put it back if you think is better.

Reply:
Updated!